examples/HowTo.md

```diff
@@ -76,29 +95,48 @@ mainWithLogging = do
     usingLoggerName "new-logger" inputLengthWithLog
 ```
 where we set up the config from the file and run `inputLengthWithLog` under the logger with name `new-logger`.
+Note that we use `inputLengthWithLog` now instead of `inputLength` because we can not just add logging
+to the function that works under `IO` monad automatically. We need to change our faction.
+And `inputLengthWithLog` is a version of `inputLength` with logging added.
 
-Let's have a closer look at how we transform the `inputLength` function to the `inputLengthWithLog` that would work with logging in the next paragraph.
+Let's have a closer look at how we transform the `inputLength` function
+to the `inputLengthWithLog` that would work with logging in the next paragraph.
 
 ## Add logging messages to function in `IO`
 
-If you want to use logging in a function that already runs in the `IO` monad, replace the monad with [`WithLoggerIO`](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#t:WithLoggerIO) and `liftIO` all your `IO` actions.
+If you want to use logging in a function that already runs in the `IO`,
+replace concrete `IO` type with polymorphic constraint
+[`WithLoggerIO m => m`](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#t:WithLoggerIO)
```

Comment:
You can write this in a cool way to add hyperlink only to `WithLogger`: [`WithLoggerIO`](some text)` m => m`